### PR TITLE
added the description of the reading order of the config.h files.

### PR DESCRIPTION
**Pull request by mtei**
## Description

I couldn't find the description of loading order of config.h files in the documents, so I added it.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/hardware_keyboard_guidelines.md [hunk 1]
```diff
@@ -61,6 +61,21 @@ This file is used by the [QMK API](https://github.com/qmk/qmk_api). It contains
...
+    * keuyboards/top_folder/sub_1/sub_2/config.h
+      * keuyboards/top_folder/sub_1/sub_2/sub_3/config.h
+        * keuyboards/top_folder/sub_1/sub_2/sub_3/sub_4/config.h
+          * users/an_user_folder/config.h
+          * keuyboards/top_folder/keymaps/an_keymap/config.h
+        * keuyboards/top_folder/sub_1/sub_2/sub_3/sub_4/post_config.h
+      * keuyboards/top_folder/sub_1/sub_2/sub_3/post_config.h
+    * keuyboards/top_folder/sub_1/sub_2/post_config.h
+  * keuyboards/top_folder/sub_1/post_config.h
+* keuyboards/top_folder/post_config.h
```
Comment (Erovia) [inline docs/hardware_keyboard_guidelines.md]:
```suggestion
* keyboards/top_folder/config.h
  * keyboards/top_folder/sub_1/config.h
    * keyboards/top_folder/sub_1/sub_2/config.h
      * keyboards/top_folder/sub_1/sub_2/sub_3/config.h
        * keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/config.h
          * users/a_user_folder/config.h
          * keyboards/top_folder/keymaps/a_keymap/config.h
        * keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/post_config.h
      * keyboards/top_folder/sub_1/sub_2/sub_3/post_config.h
    * keyboards/top_folder/sub_1/sub_2/post_config.h
  * keyboards/top_folder/sub_1/post_config.h
* keyboards/top_folder/post_config.h
```

File: docs/hardware_keyboard_guidelines.md [hunk 2]
```diff
@@ -61,6 +61,53 @@ This file is used by the [QMK API](https://github.com/qmk/qmk_api). It contains
...
+  #ifdef RGBLIGHT_ENABLE
+    #ifndef IOS_DEVICE_ENABLE
+      #define RGBLIGHT_LIMIT_VAL 200
+      #define RGBLIGHT_VAL_STEP 17
+    #else
+      #define RGBLIGHT_LIMIT_VAL 35
+      #define RGBLIGHT_VAL_STEP 4
+    #endif
+    #define RGBLIGHT_HUE_STEP 10
+    #define RGBLIGHT_SAT_STEP 17
```
Comment (fauxpark) [inline docs/hardware_keyboard_guidelines.md]: This stuff may not work if the keyboard-level config.h already defines these.

File: docs/hardware_keyboard_guidelines.md [hunk 3]
```diff
@@ -61,6 +61,53 @@ This file is used by the [QMK API](https://github.com/qmk/qmk_api). It contains
 
 All projects need to have a `config.h` file that sets things like the matrix size, product name, USB VID/PID, description and other settings. In general, use this file to set essential information and defaults for your keyboard that will always work.
 
+The `config.h` files can also be placed in sub-folders, and the order in which they are read is as follows
```
Comment (fauxpark) [inline docs/hardware_keyboard_guidelines.md]:
```suggestion
The `config.h` files can also be placed in sub-folders, and the order in which they are read is as follows:
```

File: docs/hardware_keyboard_guidelines.md [hunk 4]
```diff
@@ -61,6 +61,53 @@ This file is used by the [QMK API](https://github.com/qmk/qmk_api). It contains
...
+          * `keyboards/top_folder/keymaps/a_keymap/config.h`
+        * `keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/post_config.h`
+      * `keyboards/top_folder/sub_1/sub_2/sub_3/post_config.h`
+    * `keyboards/top_folder/sub_1/sub_2/post_config.h`
+  * `keyboards/top_folder/sub_1/post_config.h`
+* `keyboards/top_folder/post_config.h`
+
+The `post_config.h` file is for additional post-processing depending on what is specified in the `config.h` file.
+
+For example, if you define the `IOS_DEVICE_ENABLE` macro in the `keymaps/a_keymap/config.h` file as follows, you can configure the detailed settings accordingly in the `post_config.h` file.
```
Comment (fauxpark) [inline docs/hardware_keyboard_guidelines.md]:
```suggestion
The `post_config.h` file can be used for additional post-processing, depending on what is specified in the `config.h` file. For example, if you define the `IOS_DEVICE_ENABLE` macro in your keymap-level `config.h` file as follows, you can configure more detailed settings accordingly in the `post_config.h` file:
```

File: docs/hardware_keyboard_guidelines.md (hunk 2 above)
Comment (mtei) [reply docs/hardware_keyboard_guidelines.md]: The definitions have been revised and a note added.

File: docs/hardware_keyboard_guidelines.md [hunk 5]
```diff
@@ -101,11 +101,17 @@ The `post_config.h` file can be used for additional post-processing, depending o
...
+    #ifndef RGBLIGHT_HUE_STEP
+      #define RGBLIGHT_HUE_STEP 10
+    #endif
+    #ifndef RGBLIGHT_SAT_STEP
+      #define RGBLIGHT_SAT_STEP 17
+    #endif
   #endif
   ```
 
+NOTE: If you define `post_config.h` in the above example, you should not define USB_MAX_POWER_CONSUMPTION, RGBLIGHT_LIMIT_VAL, RGBLIGHT_VAL_STEP, RGBLIGHT_HUE_STEP and RGBLIGHT_SAT_STEP in keyboard-level `config.h`. And, the user-level `config.h` of this keyboard should not define USB_MAX_POWER_CONSUMPTION, RGBLIGHT_LIMIT_VAL, and RGBLIGHT_VAL_STEP.
```
Comment (fauxpark) [inline docs/hardware_keyboard_guidelines.md]:
```suggestion
?> If you define options using `post_config.h` as in the above example, you should not define the same options in the keyboard- or user-level `config.h`.
```

Comment (mtei) [reply docs/hardware_keyboard_guidelines.md]: Thanks!

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**